core/src/ast_builder/index_item/non_clustered.rs

```diff
@@ -0,0 +1,101 @@
+use {
+    super::CmpExprNode,
+    crate::{ast::IndexOperator, ast_builder::ExprNode},
+};
+
+#[derive(Clone, Debug)]
+pub struct NonClusteredNode {
+    pub index_name: String,
+}
+
+impl<'a> NonClusteredNode {
+    pub fn gt<T: Into<ExprNode<'a>>>(self, expr: T) -> CmpExprNode<'a> {
+        CmpExprNode::new(self.index_name, IndexOperator::Gt, expr.into())
+    }
+
+    pub fn lt<T: Into<ExprNode<'a>>>(self, expr: T) -> CmpExprNode<'a> {
+        CmpExprNode::new(self.index_name, IndexOperator::Lt, expr.into())
+    }
+
+    pub fn gte<T: Into<ExprNode<'a>>>(self, expr: T) -> CmpExprNode<'a> {
+        CmpExprNode::new(self.index_name, IndexOperator::GtEq, expr.into())
+    }
+
+    pub fn lte<T: Into<ExprNode<'a>>>(self, expr: T) -> CmpExprNode<'a> {
+        CmpExprNode::new(self.index_name, IndexOperator::LtEq, expr.into())
+    }
+
+    pub fn eq<T: Into<ExprNode<'a>>>(self, expr: T) -> CmpExprNode<'a> {
+        CmpExprNode::new(self.index_name, IndexOperator::Eq, expr.into())
+    }
+}
+
+pub fn non_clustered(index_name: String) -> NonClusteredNode {
+    NonClusteredNode { index_name }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        ast::IndexOperator,
+        ast_builder::{index_item::non_clustered::non_clustered, select::Prebuild, to_expr},
+        ast_builder::{index_item::IndexItem, IndexItemNode},
+    };
+
+    #[test]
+    fn test() {
+        let index_node: IndexItemNode = non_clustered("idx".to_owned()).gt("1").into();
+        let actual = index_node.prebuild().unwrap();
+        let expected = IndexItem::NonClustered {
+            name: "idx".to_owned(),
+            asc: None,
+            cmp_expr: Some((IndexOperator::Gt, to_expr("1"))),
```

Comment:
we convert `actual` into `AST` and compare with `expected`.
Then, rather than using `to_expr` which parses and translates, how about simply using AST Expr?
this below would be enough
```rust
Expr::Literal(AstLiteral::Number(1.into()))
```